Plugins/RSBot.Inventory/Subscriber/UseItemAtTrainplaceSubscriber.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using RSBot.Core.Event;
+using RSBot.Core;
+
+namespace RSBot.Inventory.Subscriber;
+
+internal class UseItemAtTrainplaceSubscriber
+{
+    private static List<string> _blacklistedItems = new();
+    private static long _lastTick;
+
+    public static void SubscribeEvents()
+    {
+        EventManager.SubscribeEvent("OnTick", OnTick);
+    }
+
+    private static void OnTick()
+    {
+        //Retry blacklisted items after 5 minutes
+        if (TimeSpan.FromTicks(DateTime.Now.Ticks - _lastTick).Minutes >= 5)
+            _blacklistedItems.Clear();
+        
+        _lastTick = DateTime.Now.Ticks;
+
```

Comment:
If the operating system is running for more than 29 days and the bot is running at that time, the tick drops to minus values, which causes a bug to use bot time. Also, the method you use is slower though in micro. There are solutions and evaluations about this in the link below.

https://blog.tedd.no/2015/08/17/comparison-of-environment-tickcount-datetime-now-ticks-and-stopwatch-elapsedmilliseconds/

**So;**

You can use Kernel.TickCount for fix the issue.

Get elapsed milliseconds: `Kernel.TickCount - _lastTick`
Get minutes if you want: ```TimeSpan.FromMilliseconds(Kernel.TickCount - _lastTick).Minutes```

Reply:
Thanks for the detailed explanation. I edited as requested.